docs/reference/troubleshooting/troubleshooting-unbalanced-cluster.asciidoc

```diff
@@ -0,0 +1,70 @@
+[[troubleshooting-unbalanced-cluster]]
+== Troubleshooting an unbalanced cluster
+
+Since 8.6 Elasticsearch is balancing shards across data tier to achieve a good compromise between
+* shard count
+* disk usage
+* ingest load forecast for write indices
+
+Please note, elasticsearch does not take into account amount or complexity of search queries when rebalancing shards.
+
+There is no guarantee that individual components are going to be evenly spread across the nodes.
+This might appear as some of the nodes have less shards allocated or using less disk
+as long as they are assigned with the shards with greater ingest load forecast.
```

Comment:
In terms of troubleshooting, I'd recommend making this more actionable. For instance, this could be step 1 of the troubleshooting process: determine whether your cluster is really unbalanced or not (from Elasticsearch's point of view). If Elasticsearch thinks it's balanced then typically there's no action needed, it will keep all nodes within the disk watermarks (link to the disk allocation docs); if the default disk watermarks are unsatisfactory then change them.

Conversely if ES thinks it's unbalanced then it will be working to balance it, but deliberately quite slowly to avoid disrupting production traffic (unless there is an urgent need, in which case it will speed up). If it struggles to reach a balanced state then try resetting the balance and/or increasing the threshold.